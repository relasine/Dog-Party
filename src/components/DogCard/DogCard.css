.DogCard {
  display: inline-block;
  max-width: 30vw;
}

.dog-card-dog {
  border-radius: 50%;
  display: block;
  margin: 0 auto;
  max-width: 20vw;
}

h2 {
  text-align: center;
}