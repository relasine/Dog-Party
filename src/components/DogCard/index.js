import React from 'react';
import './DogCard.css';
import PropTypes from 'prop-types';

const DogCard = ({ dog }) => {
  const dogSection = dog.title.split(' ')[0];

  return (
    <article className='DogCard'>
      <h2>{dog.title}</h2>
      <img className='dog-card-dog' src={dog.image} alt='A dog portrait' />
      <p>{dog.text}</p>
      <h3>{dogSection}</h3>
    </article>
  )
}

DogCard.propTypes = {
  dog: PropTypes.objectOf(PropTypes.string).isRequired
}

export default DogCard;