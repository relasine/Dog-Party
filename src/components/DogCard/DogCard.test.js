import React from 'react';
import ReactDOM from 'react-dom';
import DogCard from './index.js';
import { shallow } from 'enzyme';

describe('DogCard', () => {
  const dog = {title: 'Much Doggo, Many Bork', image: 'path/to/much/doggo/image', text: 'Doggo doggo, bork bork bork'}

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<DogCard dog={dog}/> ,div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<DogCard dog={dog}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
