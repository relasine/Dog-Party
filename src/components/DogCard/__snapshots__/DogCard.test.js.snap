// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogCard should match the snapshot 1`] = `
<article>
  <h2>
    Much Doggo, Many Bork
  </h2>
  <img
    alt="A dog portrait"
    src="path/to/much/doggo/image"
  />
  <p>
    Doggo doggo, bork bork bork
  </p>
  <h3>
    Much
  </h3>
</article>
`;
